pkg/apis/maistra/conversion/discovery_selectors.go

```diff
@@ -0,0 +1,51 @@
+package conversion
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+	v2 "github.com/maistra/istio-operator/pkg/apis/maistra/v2"
+)
+
+func populateDiscoverySelectorsValues(in *v2.ControlPlaneSpec, out map[string]interface{}) error {
+	if in.MeshConfig == nil || in.MeshConfig.DiscoverySelectors == nil {
+		return nil
+	}
+
+	untypedSlice := make([]interface{}, len(in.MeshConfig.DiscoverySelectors))
+	for index, value := range in.MeshConfig.DiscoverySelectors {
+		untypedSlice[index] = value
+	}
+	if discoverySelectors, err := sliceToValues(untypedSlice); err == nil {
+		if err := setHelmValue(out, "meshConfig.discoverySelectors", discoverySelectors); err != nil {
+			return err
+		}
+	} else {
+		return err
+	}
+
+	return nil
```

Comment:
You can simplify these conditions as below:
```suggestion
	if discoverySelectors, err := sliceToValues(untypedSlice); err == nil {
		return setHelmValue(out, "meshConfig.discoverySelectors", discoverySelectors)
	} else {
		return err
	}
```